File: iniciante/1013/1013.py
# Lê três valores inteiros digitados em uma única linha, separados por espaço
a, b, c = map(int, input().split())

# Usa a fórmula para encontrar o maior entre 'a' e 'b'
# Fórmula: (a + b + |a - b|) // 2
maiorAB = (a + b + abs(a - b)) // 2

# Agora usa a mesma fórmula para encontrar o maior entre 'maiorAB' e 'c'
maior = (maiorAB + c + abs(maiorAB - c)) // 2

# Exibe o maior valor entre os três, no formato especificado
print(f"{maior} eh o maior")

File: iniciante/1013/1013.c
#include <stdio.h>  // Biblioteca padrão para entrada e saída (ex: printf, scanf)
#include <stdlib.h> // Biblioteca necessária para usar a função abs (valor absoluto)

int main()
{
  int a, b, c, maiorAB, maior; // Declara variáveis inteiras para os três números e para os maiores valores parciais

  scanf("%d%d%d", &a, &b, &c); // Lê três inteiros fornecidos pelo usuário

  // Calcula o maior valor entre 'a' e 'b' usando a fórmula: (a + b + |a - b|) / 2
  maiorAB = (a + b + abs(a - b)) / 2;

  // Calcula o maior valor entre 'maiorAB' e 'c' usando a mesma lógica
  maior = (maiorAB + c + abs(maiorAB - c)) / 2;

  // Imprime o maior valor entre os três no formato especificado
  printf("%d eh o maior\n", maior);

  return 0; // Indica que o programa terminou corretamente
}
